src/quo2/components/list_items/contact_item.cljs

```diff
@@ -0,0 +1,56 @@
+(ns quo2.components.list-items.contact-item
+  (:require [re-frame.core :as re-frame]
+            [status-im.utils.handlers :refer [<sub]]
+            [quo2.foundations.typography :as typography]
+            [quo2.components.icon :as icons]
+            [quo2.foundations.colors :as colors]
+            [quo2.components.avatars.user-avatar :as user-avatar]
+            [quo.react-native :as rn]
+            [status-im.utils.utils :refer [get-shortened-address]]
+            [quo.platform :as platform]))
+
+(defn open-chat [chat-id]
+  (re-frame/dispatch [:dismiss-keyboard])
+  (if platform/android?
+    (re-frame/dispatch [:chat.ui/navigate-to-chat-nav2 chat-id])
+    (re-frame/dispatch [:chat.ui/navigate-to-chat chat-id]))
+  (re-frame/dispatch [:search/home-filter-changed nil])
+  (re-frame/dispatch [:accept-all-activity-center-notifications-from-chat chat-id]))
+
+(defn contact-item [item]
+  (let [{:keys [public-key]} item
+        display-name (first (<sub [:contacts/contact-two-names-by-identity public-key]))
+        photo-path   (if-not (empty? (:images item)) (<sub [:chats/photo-path public-key]) nil)]
+    [rn/touchable-opacity (merge {:style    {:margin-top         8
+                                             :margin-horizontal  8
+                                             :padding-vertical   8
+                                             :padding-horizontal 12
+                                             :border-radius      12
+                                             :flex-direction     :row
+                                             :align-items        :center}
+                                  :on-press #(open-chat public-key)})
+     [user-avatar/user-avatar {:full-name         display-name
+                               :profile-picture   photo-path
+                               :status-indicator? true
+                               :online?           true
+                               :size              :small
+                               :ring?             false}]
+     [rn/view {:style {:margin-left 8}}
+      [rn/view {:style {:flex-direction :row}}
+       [rn/text {:style (merge typography/paragraph-1 typography/font-semi-bold
+                               {:color (colors/theme-colors colors/neutral-100 colors/white)})}
+        display-name]
```

Comment:
Just personal preference, I think the code will be easier to review and maintain if put these small meaningful (in design/code level) components into separate small functions. 

Etc `display-name` for this one, it will be easier to add doc/comments to each logical components, since it seems there are lots of UI logic behind this.

Reply:
Thanks @yqrashawn. That's right.